component/admin/controllers/wrapper.php

```diff
@@ -3,225 +3,241 @@
-        }
-
-        $model = $this->getModel('wrapper_detail');
-
-        if (!$model->enableWrapperUseToAll($wrapperIds, 1)) {
-            echo "<script> alert('" . $model->getError(true) . "'); window.history.go(-1); </script>\n";
-        }
-
-        $msg = JText::_('COM_REDSHOP_USE_TO_ALL_ENABLE_SUCCESSFULLY');
-        $this->setRedirect(
-            'index.php?option=com_redshop&view=wrapper&showall=' . $showAll . $tmpl . '&product_id=' . $productId,
-            $msg
-        );
-    }
-
-    public function enable_defaultunpublish()
-    {
-        $showAll = $this->input->get('showall', '0');
-        $tmpl    = '';
-
-        if ($showAll) {
-            $tmpl = '&tmpl=component';
-        }
-
-        $productId  = $this->input->get('product_id');
-        $wrapperIds = $this->input->post->get('cid', array(0), 'array');
-
-        if (!is_array($wrapperIds) || count($wrapperIds) < 1) {
-            throw new Exception(JText::_('COM_REDSHOP_SELECT_AN_ITEM_TO_UNPUBLISH'));
-        }
-
-        $model = $this->getModel('wrapper_detail');
-
-        if (!$model->enableWrapperUseToAll($wrapperIds, 0)) {
-            echo "<script> alert('" . $model->getError(true) . "'); window.history.go(-1); </script>\n";
-        }
-
-        $msg = JText::_('COM_REDSHOP_USE_TO_ALL_DISABLE_SUCCESSFULLY');
-        $this->setRedirect(
-            'index.php?option=com_redshop&view=wrapper&showall=' . $showAll . $tmpl . '&product_id=' . $productId,
-            $msg
-        );
-    }
-
-    public function enable_defaultpublish()
-    {
-        $showAll = $this->input->get('showall', '0');
-        $tmpl    = '';
-
-        if ($showAll) {
-            $tmpl = '&tmpl=component';
-        }
-
-        $productId  = $this->input->post->get('product_id');
-        $wrapperIds = $this->input->post->get('cid', array(0), 'array');
-
-        if (!is_array($wrapperIds) || count($wrapperIds) < 1) {
-            throw new Exception(JText::_('COM_REDSHOP_SELECT_AN_ITEM_TO_UNPUBLISH'));
-        }
-
-        $model = $this->getModel('wrapper_detail');
-
-        if (!$model->enableWrapperUseToAll($wrapperIds, 1)) {
-            echo "<script> alert('" . $model->getError(true) . "'); window.history.go(-1); </script>\n";
-        }
-
-        $msg = JText::_('COM_REDSHOP_USE_TO_ALL_ENABLE_SUCCESSFULLY');
-        $this->setRedirect(
-            'index.php?option=com_redshop&view=wrapper&showall=' . $showAll . $tmpl . '&product_id=' . $productId,
-            $msg
-        );
-    }
+	/**
+	 * Method to save a record.
+	 *
+	 * @param   string  $key     The name of the primary key of the URL variable.
+	 * @param   string  $urlVar  The name of the URL variable if different from the primary key (sometimes required to avoid router collisions).
+	 *
+	 * @return  boolean          True if successful, false otherwise.
+	 * @throws  Exception
+	 */
+	public function save($key = null, $urlVar = null)
```

Comment:
@hoang-tran-redweb i see you no change any in this function so please remove it and use parent class